include/libtorrent/socket.hpp

```diff
@@ -175,6 +175,22 @@ namespace libtorrent {
 		tos_t m_value;
 	};
 
+#ifdef IP_DSCP_TRAFFIC_TYPE
+	struct dscp_traffic_type
+	{
+		explicit dscp_traffic_type(char val) : m_value(DWORD(val)) {}
+		template<class Protocol>
+		int level(Protocol const&) const { return IPPROTO_IP; }
```

Comment:
The docs seem to be saying (at least to me) that level should be `IP_DSCP_TRAFFIC_TYPE` and name should be `DSCP_TRAFFIC_TYPE`, and this would apply to either IPv4 or IPv6 sockets.

Reply:
ah, yes